src/Cli/dotnet/ToolPackage/ToolPackageDownloader.cs

```diff
@@ -295,7 +295,7 @@ private static async Task<NuGetVersion> DownloadAndExtractPackage(
             }
 
             // Extract the package
-            var nupkgDir = Path.Combine(packagesRootPath, packageId.ToString(), version.ToString());
+            var nupkgDir = Path.Combine(packagesRootPath, packageId.ToString(), version.ToNormalizedString().ToLowerInvariant());
```

Comment:
I think using APIs like VersionFolderPathResolver might be better than manually doing these calculations. 
It minimizes the number of places changes would need to be made should something change.

Unfortunately, I don't really know what other consequence that might have throughout the code, so that idea would require more investigation.,

Reply:
For a lot of them, I think they are just string modifications so I would agree it would be better to use the API. I can try replacing it where it seems relevant. Thanks!!